x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/rank/textsimilarity/TextSimilarityRankBuilder.java

```diff
@@ -165,7 +167,20 @@ public QueryPhaseRankCoordinatorContext buildQueryPhaseCoordinatorContext(int si
     }
 
     @Override
-    public RankFeaturePhaseRankShardContext buildRankFeaturePhaseShardContext() {
+    public RankFeaturePhaseRankShardContext buildRankFeaturePhaseShardContext(SearchContext searchContext) {
+
+        // check field in mapping
+        Mapper mapper;
+        try {
+            mapper = searchContext.indexShard().mapperService().mappingLookup().getMapper(field);
+        } catch (NullPointerException e) {
```

Comment:
We shouldn't use this control flow like this. Why is this throwing an NPE at all? Is it because the mapper service is null or the search context isn't the correct context?

If so, please verify the search context has the required information before making the check. Otherwise, its possible that this gets false positives (e.g. something marked as missing when it actually isn't).